src/controls/QskModelObjectBinder.cpp

```diff
@@ -0,0 +1,174 @@
+/******************************************************************************
+ * QSkinny - Copyright (C) The authors
+ *           SPDX-License-Identifier: BSD-3-Clause
+ *****************************************************************************/
+
+#include "QskModelObjectBinder.h"
+
+#include <qmetaobject.h>
+#include <qabstractitemmodel.h>
+#include <qpointer.h>
+#include <qassert.h>
+
+static inline QMetaProperty qskMetaProperty(
+    const QMetaObject* metaObject, const QByteArray& name )
+{
+    if ( name.isEmpty() )
+        return metaObject->userProperty();
+
+    const auto idx = metaObject->indexOfProperty( name );
+    return ( idx >= 0 ) ? metaObject->property( idx ) : QMetaProperty();
+}
+
+namespace
+{
+    struct Binding
+    {
+        Binding( int column, const QMetaProperty& property )
+            : property( property )
+            , column( column )
+        {
+        }
+
+        QMetaProperty property;
+        int column;
+    };
+}
+
+class QskModelObjectBinder::PrivateData
+{
+  public:
+    void updateProperties(const QModelIndex& topLeft,
+        const QModelIndex& bottomRight, const QVector< int >& roles )
+    {
+        if ( !roles.contains( Qt::EditRole ) )
+            return;
+
+        const int row = currentRowIndex.row();
+
+        if ( topLeft.row() <= row && row <= bottomRight.row() )
+        {
+            for ( auto it = bindings.begin(); it != bindings.end(); ++it )
+            {
+                const int col = it.value().column;
+
+                if( topLeft.column() <= col && col <= bottomRight.column() )
+                    updateObjectProperty( it.key(), it.value() );
+            }
+        }
+    }
+
+    void updateObjectProperty( QObject* object, const Binding& binding )
+    {
+        const auto index = model->index( currentRowIndex.row(), binding.column );
+        binding.property.write( object, index.data( Qt::EditRole ) );
+    }
+
+  public:
+    QMap< QObject*, Binding > bindings;
+    QPointer< QAbstractItemModel > model;
```

Comment:
@ovenpasta do we need to store the row as QPersistentModelIndex. When using an int the row would be valid without the model ?

Reply:
Yes, consider the following snippet:
```
auto addRow = new QskPushButton("Add row at first row",v);
    connect(addRow,&QskPushButton::clicked,[=]() {
        auto r = table->record();
        r.setValue( 0, 0 );
        r.setValue(1, "new row");
        qDebug()<< "current row before insert" << mapper->currentRow();
        table->insertRecord( 0 ,r);
        qDebug()<< "current row after insert" << mapper->currentRow();
    });
```

Output:
```
current row before insert 0
current row after insert 1
```

So technically the currentRow will adjust to the new row index after the insert.